Source/Urho3D/Core/WorkQueue.h

```diff
@@ -53,7 +53,7 @@ struct WorkItem : public RefCounted
     }
 
     /// Work function. Called with the work item and thread index (0 = main thread) as parameters.
-    void (* workFunction_)(const WorkItem*, unsigned);
+    std::function<void(const WorkItem*, unsigned)> workFunction_;

```

Comment:
[Thoughtful review needed]
How do you use it.

Reply:
https://github.com/rokups/Urho3D/blob/master/Source/Tools/Editor/SceneSettings.cpp#L189-L205

In order to put scene settings into inspector i created new class that registers settings as Urho3D attributes. Registering them as "free" attributes and using lambdas there is very convenient as it allows me to easily copy some relevant objects into scope of lambda instead of complicating code with storing of those objects and later accessing them through Serializable that is passed to getter/setter.

Reply:
I understand Attribute changes. Where do you use WorkQueue?

Reply:
Oh got confused about this one.

Used in some experiments. Again it is very convenient to use lambdas to copy some pointers into the scope of lambda and sidestep using `start_`, `end_` and `aux_` pointers. To be honest i think at some point internal engine work functions should be upgraded to use lambdas or `std::bind()` instead of those raw pointers.

Reply:
Yeah, I think so. But I don't think that this is required in _this_ PR.
Probably should be separate refactoring.